text/0000-extension-layer.md

```diff
@@ -0,0 +1,118 @@
+# Meta
+[meta]: #meta
+- Name: Add extension layer to exchange data
+- Start Date: 2023-10-09
+- Author(s): [c0d1ngm0nk3y](https://github.com/c0d1ngm0nk3y), [pbusko](https://github.com/pbusko)
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request:
+- CNB Pull Request:
+- CNB Issue:
+- Related: [RFC#0105 Support Dockerfiles](https://github.com/buildpacks/rfcs/blob/main/text/0105-dockerfiles.md)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This RFC introduces support for Extension Layers to allow data transfer between the build environment and the Kaniko execution.
+
+# Motivation
+[motivation]: #motivation
+
+This change allows extensions to possess their own layers to utilize during the generation/extend process. Additionally, it ensures that extension output does not inadvertently interfere with other extension or buildpack layers during the build, and it does not unintentionally become part of the final application image.
+
+This would allow distroless run images to be extended.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This follows up on RFC-0105 and proposes that during the execution of the extension's `./bin/generate`, an extension is allowed to write arbitrary data to the `context` folder within its exclusive layer. This data then becomes accessible during the execution of the `extend` phase via Kaniko build context. The content of these extension-specific layers is ignored at build and launch time, it serves only the extension phase.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+- New root directory `/layers-ext` is introduced which contains extension layers.
+- Before execution of the `./bin/generate`, the lifecycle will create a distinct writable layer `/layers-ext/<extension-id>` for each extension which passed detection.
+- The `/layers-ext/<extension-id>` is provided to the `./bin/generate` as `<output>` directory.
```

Comment:
Today the spec says "If an image extension is missing /bin/generate, the image extension root /generate directory MUST be treated as a pre-populated <output> directory." This would allow for pre-populated context directories as well. I wonder if there's a case for decoupling the pre-populated context directories from a missing /bin/generate.

Reply:
I would say that the pre-populated context directory as part of the `/generate` directory is a valid use-case. Since no `./bin/generate` executable will be called it would be basically the only way to provide a context.